scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -90,3 +90,12 @@ function validate_resource_files() {
   done
 }
 
+function validate_broker_resources() {
+
+  ${DAPPER_SOURCE}/bin/subctl gather --kubeconfig ${KUBECONFIGS_DIR}/kind-config-${cluster} --dir $out_dir
+
+  validate_resource_files $SUBMARINER_BROKER_NS 'endpoints.submariner.io' 'Endpoint'
+  validate_resource_files $SUBMARINER_BROKER_NS 'clusters.submariner.io' 'Cluster'
+  validate_resource_files all 'serviceimports.multicluster.x-k8s.io' 'ServiceImport'
+  validate_resource_files all 'endpointslices.discovery.k8s.io' 'EndpointSlice'
+}
```

Comment:
We don't need to run `subctl gather` again specifically for the broker.  You should be able to do:

```
function validate_broker_resources() {
  validate_resource_files $SUBMARINER_BROKER_NS 'endpoints.submariner.io' 'Endpoint'
  validate_resource_files $SUBMARINER_BROKER_NS 'clusters.submariner.io' 'Cluster'
  validate_resource_files $SUBMARINER_BROKER_NS 'serviceimports.multicluster.x-k8s.io' 'ServiceImport'
  validate_resource_files $SUBMARINER_BROKER_NS 'endpointslices.discovery.k8s.io' 'EndpointSlice'
}

function test_subctl_gather() {
  ...
  # Broker
  with_context "$broker" validate_broker_resources
}
```

Also the E2E failed in the broker validation b/c it uses "cluster1" in the file names instead of "broker". We can modify `validate_resource_files` to take an additional optional arg:

```
function validate_resource_files() {
  ...
  local cluster_name=$5

  if [[ $cluster_name == "" ]]; then
    cluster_name=${cluster}
  fi
  ...
  for name in "${names_array[@]}"; do
    file=$out_dir/${cluster_name}_${short_res}_${ns}_${name}.yaml
    ...
  done
}
```

Then to call it

`validate_resource_files $SUBMARINER_BROKER_NS 'endpoints.submariner.io' 'Endpoint' '' 'broker'`